docs/spec/auth/vesting.md

```diff
@@ -0,0 +1,98 @@
+## Vesting
+
+### Intro and Requirements
+
+This paper specifies changes to the auth and bank modules to implement vested accounts for the Cosmos Hub. 
+The requirements for this vested account is that it should be capable of being initialized during genesis with
+a starting balance X coins and a vesting blocknumber N. The owner of this account should be able to delegate to validators and vote,
+however they cannot send their coins to other accounts until the account has fully vested. Thus, the bank module's `MsgSend` handler 
+should error if a vested account is trying to send an amount before block N.
+
+### Implementation
+
+##### Changes to x/auth Module
+
+The `Account` interface will specify both the Account type and any parameters it needs.
+
+```go
+// Account is a standard account using a sequence number for replay protection
+// and a pubkey for authentication.
+type Account interface {
+    Type() string // returns the type of the account
+
+	GetAddress() sdk.AccAddress
+	SetAddress(sdk.AccAddress) error // errors if already set.
+
+	GetPubKey() crypto.PubKey // can return nil.
+	SetPubKey(crypto.PubKey) error
+
+	GetAccountNumber() int64
+	SetAccountNumber(int64) error
+
+	GetSequence() int64
+	SetSequence(int64) error
+
+	GetCoins() sdk.Coins
+    SetCoins(sdk.Coins) error
+    
+    // Getter and setter methods for account params
+    // It is upto handler to use these appropriately
+    GetParams() map[string]interface{}
+    SetParams(map[string]interface{}) error
+}
```

Comment:
shouldn't this be more like `GetParams([]byte) []byte` and `SetParams([]byte, []byte)`
p.s. golang maps are a non-deterministic type meaning we can't use them in the state-machine